crates/polars-core/src/chunked_array/temporal/date.rs

```diff
@@ -32,34 +32,20 @@ impl DateChunked {
 
     /// Convert from Date into String with the given format.
     /// See [chrono strftime/strptime](https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html).
-    pub fn to_string(&self, format: &str) -> StringChunked {
-        let mut ca: StringChunked = self.apply_kernel_cast(&|arr| {
-            let mut buf = String::new();
-            let mut mutarr = MutablePlString::with_capacity(arr.len());
-
-            for opt in arr.into_iter() {
-                match opt {
-                    None => mutarr.push_null(),
-                    Some(v) => {
-                        buf.clear();
-                        let datefmt = date32_to_date(*v).format(format);
-                        write!(buf, "{datefmt}").unwrap();
-                        mutarr.push_value(&buf)
-                    },
-                }
-            }
-
-            mutarr.freeze().boxed()
-        });
-        ca.rename(self.name());
-        ca
+    pub fn to_string(&self, format: &str) -> PolarsResult<StringChunked> {
+        let datefmt_f = |ndt: NaiveDate| ndt.format(format);
+        self.try_apply_to_buffer_generic(|val, buf| {
+            let ndt = date32_to_date(val);
+            write!(buf, "{}", datefmt_f(ndt))
+        })
+        .map_err(|_| polars_err!(ComputeError: "cannot format Date with format '{}'", format))
```

Comment:
I hope `try_apply_to_buffer_generic` is OK because this diff is really satisfying 😋